### PR TITLE
Updates dynamic IP address assignment configuration with whereabouts …

### PR DESCRIPTION
**Pull request by stevsmit**
…docs to include manual creation of YAML manifest

Version(s):
4.10+

Issue:
https://issues.redhat.com/browse/OSDOCS-6917

Link to docs preview:
https://62511--docspreview.netlify.app/openshift-enterprise/latest/networking/multiple_networks/configuring-additional-network.html#nw-multus-creating-whereabouts-reconciler-daemon-set_configuring-additional-network

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

**Comment by weliang1** [comment]
The section of “Dynamic IP address assignment configuration with Whereabouts” is about how a POD get a IP address through: 1. An additional network from a YAML manifest,  or 2: configuring an additional network through the Cluster Network Operator.

But the bug 
https://issues.redhat.com/browse/OSDOCS-6917 is about when the Whereabouts-reconciler daemonset will be created, it isn't about how a POD get a dynamic IP address.

How about add a new section for your new comments about  creating a Whereabouts-reconciler daemonset, and include example of configuring an additional network through the Cluster Network Operator in the section of “Dynamic IP address assignment configuration with Whereabouts”?

File: modules/nw-multus-ipam-object.adoc [hunk 1]
```diff
@@ -240,3 +240,52 @@ Whereabouts can be used for both IPv4 and IPv6 addresses.
...
+== Creating a Whereabouts-reconciler daemonset 
+
+The Whereabouts-reconciler is responsible for managing dynamic IP address assignments for the pods within a cluster using the Whereabouts IP Address Management (IPAM) solution. It ensures that each pods gets a unique IP address from the specified IP address range. It also handles IP address releases when pods are deleted or scaled down.
+
+[NOTE]
+====
+You can also use a `NetworkAttachmentDefinition` custom resource for dynamic IP address assignment. 
+====
+
+The Whereabouts-reconciler daemonset is automatically created when you configure an additional network through the Cluster Network Operator. It is not automatically created when you configure an additional network from a YAML manifest.
```
Comment (jab-rh) [inline modules/nw-multus-ipam-object.adoc:254]:
```suggestion
The Whereabouts-reconciler daemon set is automatically created when you configure an additional network through the Cluster Network Operator. It is not automatically created when you configure an additional network from a YAML manifest. 
```

File: modules/nw-multus-ipam-object.adoc [hunk 2]
```diff
@@ -240,3 +240,52 @@ Whereabouts can be used for both IPv4 and IPv6 addresses.
...
+
+The Whereabouts-reconciler daemonset is automatically created when you configure an additional network through the Cluster Network Operator. It is not automatically created when you configure an additional network from a YAML manifest. 
+
+To trigger the deployment of the Whereabouts-reconciler daemonset, you must manually create a `whereabouts-shim` network attachment by editing the Cluster Network Operator `config.yaml` file.
+
+Use the following procedure to deploy the Whereabouts-reconciler daemonset. 
+
+.Procedure
+
+. Update your Cluster Network Operator `config.yaml` file to add the `whereabouts-shim` network attachment definition. For example:
```
Comment (jab-rh) [inline modules/nw-multus-ipam-object.adoc:262]: There's actually no `config.yaml` for this; The CNO configuration is what's considered a _singleton_, the object is created as part of cluster installation. Well, someone can modify the defaults for it, but that won't help someone here.

To do this, you'd need to use `oc edit` on the custom resource. I think this is what we're looking for:

`oc edit network.operator.openshift.io cluster`

And the `additionalNetworks` array will be in there, possibly populated with objects, if the user has already added some this way.

This approach is basically what's in here: `modules/nw-kube-proxy-configuring.adoc`

File: modules/nw-multus-ipam-object.adoc [hunk 3]
```diff
@@ -240,3 +240,52 @@ Whereabouts can be used for both IPv4 and IPv6 addresses.
 ifdef::sr-iov[]
 :!sr-iov:
 endif::[]
+
+[id="nw-multus-creating-whereabouts-reconciler-daemonset_{context}"]
+== Creating a Whereabouts-reconciler daemonset
```
Comment (jab-rh) [inline modules/nw-multus-ipam-object.adoc:245]: I wonder if there's a hyphen for this or not. We've never talked about reconcilers before. No hits on a docs search.

**Review by jab-rh — COMMENTED** [review]
@stevsmit, left a few comments. Thanks for working on this!

**Comment by stevsmit** [comment]
@jab-rh please take another look!

File: modules/nw-multus-ipam-object.adoc [hunk 4]
```diff
@@ -240,3 +240,61 @@ Whereabouts can be used for both IPv4 and IPv6 addresses.
...
+       "ipam": {
+         "type": "whereabouts"
+       }
+      }
+    type: Raw
+----
+
+. Save the file and exit the text editor.
+
+. Enter the following command to apply the `whereabouts-shim` to your Cluster Network Operator configuration:
```
Comment (weliang1) [inline modules/nw-multus-ipam-object.adoc:295]: I do not think you need step 4 any more after you using step 1

Comment (weliang1) [reply modules/nw-multus-ipam-object.adoc:295]: For step 4, you can add verification step:
`[weliang@weliang openshift-tests-private]$ oc get all -n openshift-multus | grep whereabouts-reconciler
pod/whereabouts-reconciler-jnp6g                   1/1     Running   0          6s
pod/whereabouts-reconciler-k76gg                   1/1     Running   0          6s
pod/whereabouts-reconciler-k86t9                   1/1     Running   0          6s
pod/whereabouts-reconciler-p4sxw                   1/1     Running   0          6s
pod/whereabouts-reconciler-rvfdv                   1/1     Running   0          6s
pod/whereabouts-reconciler-svzw9                   1/1     Running   0          6s
daemonset.apps/whereabouts-reconciler          6         6         6       6            6           kubernetes.io/os=linux   6s
[weliang@weliang openshift-tests-private]$ 
`

Comment (jab-rh) [reply modules/nw-multus-ipam-object.adoc:295]: @weliang1, I like this, thanks!

Comment (stevsmit) [reply modules/nw-multus-ipam-object.adoc]: @weliang1 @jab-rh I think we're finally good here. Thanks for the review!

**Comment by stevsmit** [comment]
Requesting merge review due to content change.

File: modules/nw-multus-ipam-object.adoc [hunk 5]
```diff
@@ -240,3 +240,74 @@ Whereabouts can be used for both IPv4 and IPv6 addresses.
...
+       "ipam": {
+         "type": "whereabouts"
+       }
+      }
+    type: Raw
+----
+
+. Save the file and exit the text editor.
+
+. Verify that the `whereabouts-reconciler` resources deployed successfully by running the following command:
```
Comment (JoeAldinger) [inline modules/nw-multus-ipam-object.adoc:295]: Is the `whereabouts-reconciler` a resource or a daemon set?

**Review by JoeAldinger — COMMENTED** [review]
One question, otherwise looks good to me.

**Pull request merged**